packages/microfiche/src/microfiche.js

```diff
@@ -0,0 +1,113 @@
+function assign(a, b) {
+  for (var key in a) {
+    if (a[key] !== undefined) {
+      b[key] = a[key]
+    }
+  }
+
+  return a
+}
+
+export class Database {
+  constructor() {
+    this.head = new Changeset({})
+  }
+
+  get() {
+    return this.head.get.apply(this.head, arguments)
+  }
+
+  each() {
+    return this.head.each.apply(this.head, arguments)
+  }
+
+  rollback() {
+    this.head = this.head.rollback()
+  }
+
+  transact(fn) {
+    let draft = this.head.spawn()
+
+    fn(draft)
+
+    this.head = draft
+  }
+
+  compress() {
+    this.head = this.head.compress()
+  }
+}
+
+export class Changeset {
+  constructor(last) {
+    this.last = last
+    this.facts = Object.create(last)
+  }
+
+  compress() {
+    return new Changeset(assign({}, this.facts))
```

Comment:
will `assign({}, this.facts)` do anything? I see `for (var key in a) {` in `assign`. which will iterate over an empty object thus doing nothing, then it will return the empty object. am i missing something?